src/GitVersion.Core.Tests/IntegrationTests/MainlineDevelopmentMode.cs

```diff
@@ -426,6 +426,32 @@ public void MergingFeatureBranchThatIncrementsMinorNumberIncrementsMinorVersionO
             fixture.AssertFullSemver("1.1.0", currentConfig);
         }
 
+        [Test]
+        public void VerifyMergeRemoteMainIncrementsAllCommits()
+        {
+            using var remote = new EmptyRepositoryFixture();
+
+            remote.Repository.MakeACommit("1");
+            remote.MakeATaggedCommit("1.0.0");
+
+            using var local = remote.CloneRepository();
+
+            local.Repository.MakeACommit("l.1");
+
+            remote.Repository.MakeACommit("r.1");
+            remote.Repository.MakeACommit("r.2");
+            remote.Repository.MakeACommit("r.3");
+
+            local.AssertFullSemver("1.0.1", config);
+            remote.AssertFullSemver("1.0.3", config);
+
+            Commands.Pull((Repository)local.Repository, Generate.SignatureNow(), new PullOptions());
```

Comment:
Can you specify the pull strategy explicitly so we are certain a `merge` and not a `ffwd` or `rebase` is performed here?

Reply:
Of course. Also added assertion that the tip is indeed a merge.

Reply:
Hmm. I'm not sure how this is supposed to be done. I added a commit to the PR. Not sure what to do about this change request thing I'm commenting on right now.

Reply:
Changing a reviewed line dismisses the review, forcing me to take another look. So this is working just like intended. 😄  👍🏼